homeassistant/components/zeroconf/__init__.py

```diff
@@ -382,25 +405,23 @@ async def _process_service_update(
                     # likely bad homekit data
                     return
 
-        if ATTR_NAME in info:
-            lowercase_name: str | None = info[ATTR_NAME].lower()
+        if info.name:
+            lowercase_name: str | None = info.name.lower()
         else:
             lowercase_name = None
 
-        if "macaddress" in info[ATTR_PROPERTIES]:
```

Comment:
I should clean this up a bit in a new PR.

Reply:
I wanted to keep the changes to a minimum.
If I happen to miss it, it will definitely get picked up when we enable the warning or when we remove `__getattr__`

Reply:
I agree.  I'll open a followup after this merges to do a touch up here

Reply:
https://github.com/home-assistant/core/pull/60293